### PR TITLE
kaniko demo using build-definitions script

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: hack/chains/kaniko-demo.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/bin/bash
 
 source $(dirname $0)/_helpers.sh
-set -ue
```
Comment (lcarva) [inline hack/chains/kaniko-demo.sh:4]: Why remove this?

File: hack/chains/kaniko-demo.sh [hunk 2]
```diff
@@ -40,10 +39,13 @@ tkn task start kaniko-chains \
 title "Wait a few seconds for chains finalizers to complete"
 sleep 10
 
-# This will use cosign to verify the new build
-$SCRIPTDIR/kaniko-cosign-verify.sh
+title "Create and run the verification task"
+oc create -f verify-attestation-task.yaml
```
Comment (lcarva) [inline hack/chains/kaniko-demo.sh:43]: Consider using `oc apply` instead so we can run the script multiple times.

File: hack/chains/kaniko-demo.sh (hunk 1 above)
Comment (joejstuart) [reply hack/chains/kaniko-demo.sh]: I think I did while debugging. Adding it back.

File: hack/chains/verify-attestation-task.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: verify-attestation-signature
```
Comment (simonbaird) [inline hack/chains/verify-attestation-task.yaml:4]: I'm wondering if it is too soon to define this task over in https://github.com/redhat-appstudio/build-definitions/tree/main/tasks ?

File: hack/chains/README.md [hunk 4]
```diff
@@ -33,10 +33,6 @@ demonstrate chains.)
     cd hack/chains
     ./create-signing-secret.sh
 
-### Apply some CA cert related hacks to make SSL work for the internal registry
-
-    ./trust-local-cert.sh
-
```
Comment (simonbaird) [inline hack/chains/README.md:39]: I reworded this in #188 FYI so we might get a conflict. See if you like my change there and if it makes it worthwhile to keep mentioning this script in the readme.

**Comment by simonbaird** [comment]
Looks cool! I still have the wish to support run the "old" demo as well as the new one using a taskrun, but maybe I don't care so much now.

File: hack/chains/kaniko-demo.sh [hunk 5]
```diff
@@ -40,10 +40,13 @@ tkn task start kaniko-chains \
...
+oc apply -f verify-attestation-task.yaml
+tkn task start verify-attestation-signature \
+  --use-param-defaults \
+  --showlog
 
 pause
 
 # This will use rekor-cli to verify the new build
-$SCRIPTDIR/rekor-verify-taskrun.sh
+#$SCRIPTDIR/rekor-verify-taskrun.sh
```
Comment (simonbaird) [inline hack/chains/kaniko-demo.sh:52]: Are you planning to add a taskrun for this part as well?

**Comment by simonbaird** [comment]
See also @zregvart's  #197 and the related task definition PR which I think is a stronger version of this.

**Pull request closed without merging**